app/types/app-extension.d.ts

```diff
@@ -0,0 +1,61 @@
+import { IResolve } from './app-paths'
+
+export interface IndexAPI {
+  ctx: Record<string, unknown>
+  extId: string
+  prompts: Record<string, unknown>
+  resolve: IResolve,
+  appDir: string,
+  // __hooks: IndexAPIHooks,
+  getPersistentConf: () => Record<string, unknown>,
+  setPersistentConf: (cfg: Record<string, unknown>) => void,
+  mergePersistentConf: (cfg: Record<string, unknown>) => void,
+  compatibleWith: (packageName: string, semverCondition?: string) => void,
+  hasPackage: (packageName: string, semverCondition?: string) => boolean,
+  hasExtension: (extId: string) => boolean,
+  getPackageVersion: (packageName: string) => string|undefined
+  extendQuasarConf: (fn: Function) => void,
+  chainWebpack: (fn: Function) => void,
+  extendWebpack: (fn: Function) => void,
+  chainWebpackMainElectronProcess: (fn: Function) => void,
+  extendWebpackMainElectronProcess: (fn: Function) => void,
+  chainWebpackWebserver: (fn: Function) => void,
+  extendWebpackWebserver: (fn: Function) => void,
+  registerCommand: (commandName: string, fn: Function) => void,
```

Comment:
It's always better to avoid `Function`, probably a better type would be `(payload: {args: unknown[], params: Record<string, string>}): Promise<void> | void` or something similar